### ISSUE TITLE
Chat Search

### ISSUE THREAD

**Issue by gaboesquivel** [bitcashorg/masterbots#183]
Restore chat search box

- use `?query={encodeQuery(queryString)}` with shallow link as in search,
- if it fails to refresh then do the same workaround for now, both should follow exact implementation patter
- use supabase server client

depends on #166

**Comment by sheriffjimoh** [comment]
@AndlerRL Is this up?  can I get more info? this is the search in the chat page right ?

**Comment by AndlerRL** [comment]
> @AndlerRL Is this up?  can I get more info? this is the search in the chat page right ? 
> 
> 

Yes, it is up @sheriffjimoh . This is for search functionalities on alpha.masterbots.ai for this we want to enhance the search to be SSR instead of client side (which is the current version of MasterBots.ai). With this task we want to restore, it seems the queries are not optimized to give a good result. The search consists of keywords on messages and it is failing because it is not finding the role of each message on thread. @gaboesquivel may have more info about it, he has a initial version for search, but the issue right now is that the DB doesn't have data on alpha version.

Check the main issue related to this, there is a little more on context.

**Comment by sheriffjimoh** [comment]
> Restore chat search box
> 
> * use `?query={encodeQuery(queryString)}` with shallow link as in search,
> * if it fails to refresh then do the same workaround for now, both should follow exact implementation patter
> * use supabase server client
> 
> depends on #166

New approach should use SSR with supabase? Correct?

**Comment by AndlerRL** [comment]
@sheriffjimoh — Yes, we have supabase backend set up. main and develop are hearing from genql (like usual) but the rest of the branches use supabase more directly.

**Comment by Bran18** [comment]
@AndlerRL another one from a different branch than the current one. But this one is co-related to the search improvement we can do in our current path. 

We should close this one. We only need to create one issue for **Searching improvements** in our current branch and path

**Comment by AndlerRL** [comment]
Yes, we have to update the search requirements indication. Now the search will go through the ICL vectoring instead of other strategies. I will be closing as not planned, the ICL by product design will handle not only the search bar but also search of examples for new chats inputs.

**Comment by AndlerRL** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Brandon Development Support - October 2024](https://trello.com/c/SzgyUblj/239-brandon-development-support-october-2024)
